src/EtherFiRewardsRouter.sol

```diff
@@ -0,0 +1,73 @@
+pragma solidity ^0.8.24;
+
+import "@openzeppelin-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import "@openzeppelin-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "./RoleRegistry.sol";
+
+contract EtherFiRewardsRouter is OwnableUpgradeable, UUPSUpgradeable  {
+    using SafeERC20 for IERC20;
+
+    address public immutable treasury;
+    address public immutable liquidityPool;
+    RoleRegistry public immutable roleRegistry;
+
+    bytes32 public constant ETHERFI_ROUTER_ADMIN = keccak256("ETHERFI_ROUTER_ADMIN");
```

Comment:
would like to note that we should upgrade this contract to remove this role in the future